src/Libraries/Microsoft.Extensions.DataIngestion/Chunkers/SemanticSimilarityChunker.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics.Tensors;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.AI;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.DataIngestion.Chunkers;
+
+/// <summary>
+/// Splits a <see cref="IngestionDocument"/> into chunks based on semantic similarity between its elements.
+/// </summary>
+public sealed class SemanticSimilarityChunker : IngestionChunker<string>
```

Comment:
@KrystofS Is it possible to solve this problem with other approaches? Should we provide some additional details in the comments? Like the name of the algorithm it implements so if we add more implementations in the future, the users can easily choose the right implementation?